docs/What_Is_A_Custom_Device.md

```diff
@@ -0,0 +1,172 @@
+
+&nbsp;&nbsp;&nbsp; **Note:** You can use the **[.NET API](https://www.ni.com/documentation/en/veristand/latest/manual/veristand-net-reference/)** to programmatically modify the system definition.
+
+#### Main Page
+
+The Custom Device Template Tool creates Main Page.vi. After the custom device has been added to the system definition, the main page runs whenever the operator clicks on the custom device’s top-level item in the System Explorer configuration tree. The following image displays the top-level item.
+
+![](images/Picture6.png)
+<br />
+
+#### Engine
+
+The Custom Device Template Tool creates the <span style="color:green">*<Custom Device Name> RT Driver.vi.*</span> This defines the behavior of the custom device on the **[execution host](http://zone.ni.com/reference/en-XX/help/372846B-01/veristand/veristand_glossary/#execution_host)**. The RT Driver VI runs on the execution host regardless of the target’s operating system. VeriStand deploys the engine when the operator runs the project from VeriStand or when the system definition is deployed using the VeriStand Execution API.
+
+VeriStand 2009 did not support the VeriStand Engine on VxWorks operating systems. Starting with VeriStand 2010, if you want to support VxWorks targets such as Compact RIO, you must compile the engine library for VxWorks. PharLap and Windows engines do not require additional compilation.
+
+The engine runs after the custom device deployed to the execution host. You can usually add initialization, steady-state, and shutdown code to the engine template. There aren't any hard boundaries on what code you can put into the engine, but each additional code that is added can increase the size of the engine, and the time required to deploy your system.
+
+Each of the five prebuilt custom devices has a different engine VI. Each engine VI executes at a different time with respect to other VeriStand components. The timing requirements of a custom device, and thus the type of device selected, are functions of when the device needs to execute with respect to other VeriStand Engine components.
+
+Not all requirements can be satisfied by one of the five types of prebuilt custom devices. Some custom devices will require multiple engine libraries, such as (to support different real-time operating systems for example). **[NI VeriStand – Set Custom Device Driver VI](https://zone.ni.com/reference/en-XX/help/372846M-01/veristandmerge/vs_set_custom_device_drivers/)** allows you to programmatically change the driver library for a custom device. 
+Some custom devices use the prebuilt template as a launching pad for multiple parallel processes or complex frameworks.
+For more information, refer to **Beyond the Template Frameworks**.
```

Comment:
We should make the **Beyond the Template Frameworks** visible somehow that is should be a hiperlink in the future. Maybe make it Red and add a note (insert hiperlink) or something, to not forget to connect it when the page is ready.